stdlib/Distributed/test/distributed_exec.jl

```diff
@@ -1682,6 +1682,30 @@ let (h, t) = Distributed.head_and_tail(Int[], 0)
     @test collect(t) == []
 end
 
+
+# issue #34333 test remotecall with Function-like objects
+let
+    id = 1
+
+    struct Polynomial{R}
+        coeffs::Vector{R}
+    end
+
+    function (p::Polynomial)(x)
+        v = p.coeffs[end]
+        for i = (length(p.coeffs)-1):-1:1
+            v = v*x + p.coeffs[i]
+        end
+        return v
+    end
+
+    p = Polynomial([1,10,100])
+    args = 2
+    result = p(2)
+    @test fetch(remotecall(p, id, args)) == result
+    @test fetch(remotecall_wait(p, id, args)) == result
+end
```

Comment:
This is more elaborate than necessary, and doesn't test remote workers. `remotecall(Float64, id, 1)` would suffice.

Reply:
@JeffBezanson  Simplified the tests  in bcdf987, does `remotecall_fetch` test the remote workers case?

Reply:
The case of `id = 1` happens to already work (calls on the local process don't need to send a message so they go through a different path), so it needs to be a different id for the test to be meaningful.

Reply:
@JeffBezanson changed from 1 to id_other in 2ae9479, since id_other is defined as `id_other = filter(x -> x != id_me, procs())[rand(1:(nprocs()-1))]`